### PR TITLE
Add angular drivetrain tests

### LINKED ISSUE #149: Finalize the Angular Drivetrain Test
The angular drivetrain hasn't been finalized nor integrated into the analysis tests.

### PR DESCRIPTION
**Pull request by Piphi5**
This works with the current drivetrain simulation model in wpilib although it will probably be updated soon.
Fixes #149

### CODE REVIEW

**Comment by Piphi5** [comment]
Now uses angular gyro data rather than velocities.

**Comment by Piphi5** [comment]
Should there be unit tests for the angular analysis or does the integration test work fine?

File: sysid-application/src/main/native/cpp/analysis/AnalysisManager.cpp [hunk 1]
```diff
@@ -247,17 +258,24 @@ static void PrepareAngularDrivetrainData(
...
     for (auto&& pt : it->second) {
-      pt[kVoltageCol] = 2 * std::copysign(pt[kVoltageCol], pt[kAngularRateCol]);
       pt[kLPosCol] *= factor;
       pt[kRPosCol] *= factor;
+
+      // Store the summarized voltages and velocities in the left column
+      pt[kLVoltageCol] =
+          (std::copysign(pt[kLVoltageCol], pt[kAngularRateCol]) +
+           std::copysign(pt[kRVoltageCol], pt[kAngularRateCol])) /
+          2;
```
Comment (calcmogul) [inline sysid-application/src/main/native/cpp/analysis/AnalysisManager.cpp:268]: The comment is wrong. The velocity isn't being stored in the left column.

**Review by calcmogul — CHANGES_REQUESTED** [review]
Could you post a screenshot of the changes to GUI created by DisplayFeedforwardGains() and some combined plots from an angular test?

**Comment by calcmogul** [comment]
There should be unit tests for all types of analysis, both linear and angular.

**Comment by Piphi5** [comment]
Here's the added spacing for track width:
image
Here's a combined plot:
image

File: sysid-application/src/main/native/include/sysid/analysis/AngularDriveSim.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * purposes.
+ */
+class AngularDriveSim {
+ public:
+  /**
+   * @param Ks              Static friction gain.
+   * @param Kv              Velocity gain.
+   * @param Ka              Acceleration gain.
+   * @param initialPosition Initial elevator position.
+   * @param initialVelocity Initial elevator velocity.
```
Comment (calcmogul) [inline sysid-application/src/main/native/include/sysid/analysis/AngularDriveSim.h:23]: Is this linear or angular position/velocity? The model states should be documented.

**Review by calcmogul — CHANGES_REQUESTED** [review]

File: sysid-application/src/main/native/include/sysid/analysis/AngularDriveSim.h (hunk 2 above)
Comment (Piphi5) [reply sysid-application/src/main/native/include/sysid/analysis/AngularDriveSim.h]: Is the new explanation better?

Comment (calcmogul) [reply sysid-application/src/main/native/include/sysid/analysis/AngularDriveSim.h]: Do Kv and Ka also use angular units here?

Comment (Piphi5) [reply sysid-application/src/main/native/include/sysid/analysis/AngularDriveSim.h]: No, they are linear. Currently, the model takes the linear gains converted into angular gains, and then when generating the data, it spits out the linear gains again (effectively canceling out the conversions). Would it just be better to start with the linear gains and just spit those out?

Comment (calcmogul) [reply sysid-application/src/main/native/include/sysid/analysis/AngularDriveSim.h]: The linear ones are fine. Just document the units.

**Review by calcmogul — APPROVED** [review]

**Pull request merged**